compiler/cpp/src/thrift/generate/t_py_generator.cc

```diff
@@ -871,7 +871,12 @@ void t_py_generator::generate_py_struct_definition(ostream& out,
       }
 
       if (is_immutable(tstruct)) {
-        if (gen_newstyle_ || gen_dynamic_) {
+        if (gen_enum_ && type->is_enum()) {
+          indent(out) << "super(" << tstruct->get_name() << ", self).__setattr__('"
```

Comment:
this is probably too much magic to my preference, makes me start to prefer the approach in #2824 again 😅

Reply:
Yeah I was thinking about this and then I realized that the fix was actually broken because I wasn't running one of the tests that I added.... 😓 . So actually this fix is needed to make sure existing users aren't broken.

```python
if __name__ == "__main__":
    serialization_deserialization_enum_test()
    serialization_deserialization_string_test
```

Reply:
so this code:
```py

    def __init__(self, whatOp=None, why=None, who=None,):
        super(TestException, self).__setattr__('whatOp', whatOp)
        super(TestException, self).__setattr__('why', why if hasattr(why, 'value') else shared_types.ttypes.SharedEnum.__members__.get(why))
        super(TestException, self).__setattr__('who', who if hasattr(who, 'value') else TestEnum.__members__.get(who))
```
calls the parent's `__setattr__`, doesn't that essentially set those fields on the parent instead? I guess it still "works" as when accessing them you just inherit from the parent, but this _feels_ wrong to me.

Reply:
The code that generates `super(TestException, self).__setattr__('whatOp', whatOp)` already exists before my change and I am extending this pattern. In python, [super](https://docs.python.org/3/library/functions.html#super) is implemented as part of the binding process for explicit dotted attribute lookups such as __setattr__ and such. So I don't think this is a concern because this is how Python does things under the hood.